### PR TITLE
[com_actionlogs] - log checkin actions

### PR DESCRIPTION
**Pull request by alikon**
### Summary of Changes
keep track of check-in actions

### Testing Instructions
-  Maintenance: Global Check-in
- Check-in a checked out item

### Expected result

screenshot from 2018-10-07 13-12-47

### Actual result
not logged

### CODE REVIEW

**Comment by jsubri** [comment]
I'm using staging but I can't get it working despite I've inserted manually the row in #_action_logs_extensions. I noticed their is a 2h difference between this PR and committed files in Staging. Are you in sync?

**Review by mbabker — DISMISSED** [review]
Needs update SQL files as well as changing the `com_actionlogs` default params in the `joomla.sql` files.

**Comment by alikon** [comment]
the changes requested

**Comment by jsubri** [comment]
Not sure what is the practice for things intwined between Beta cycle.
This is difficult to test using patch_tester for Upgraded site that are already to 3.9B4 as this PR need other committed stuff already in staging. Can we have the supplemental files in administrator/components/com_admin... ?  e.g not only for fresh install

**Comment by mbabker** [comment]
The additions to the `#__action_logs_extensions` table should be their own SQL files.  The changes to the parameter values shouldn't as that changes an existing user configuration (so at most for someone who is already on beta track, they'd just need to go in and re-save the component config adding the new component; keep in mind https://github.com/joomla/joomla-cms/issues/21024 has yet to have an "official" response and as such the project does not support upgrades between non-stable releases).

File: administrator/components/com_admin/sql/updates/sqlazure/3.9.0-2018-05-05.sql [hunk 1]
```diff
@@ -70,7 +70,9 @@ SELECT 15, 'com_tags'
 UNION ALL
 SELECT 16, 'com_templates'
 UNION ALL
-SELECT 17, 'com_users';
+SELECT 17, 'com_users',
```
Comment (Quy) [inline administrator/components/com_admin/sql/updates/sqlazure/3.9.0-2018-05-05.sql]: Remove comma

File: administrator/language/en-GB/en-GB.plg_actionlog_joomla.ini [hunk 2]
```diff
@@ -35,6 +35,7 @@ PLG_ACTIONLOG_JOOMLA_TYPE_TEMPLATE="template"
 PLG_ACTIONLOG_JOOMLA_TYPE_USER="user"
 PLG_ACTIONLOG_JOOMLA_TYPE_USER_GROUP="user group"
 PLG_ACTIONLOG_JOOMLA_TYPE_USER_NOTE="user note"
+PLG_ACTIONLOG_JOOMLA_USER_CHECKIN="User <a href='{accountlink}'>{username}</a> performed check-in to table {table}"
```
Comment (brianteeman) [inline administrator/language/en-GB/en-GB.plg_actionlog_joomla.ini]: ==> performed a check in to table

ref https://developer.joomla.org/en-gb-user-interface-text-guidelines/action-or-description.html

Comment (alikon) [reply administrator/language/en-GB/en-GB.plg_actionlog_joomla.ini]: fixed, thanks

**Review by brianteeman — APPROVED** [review]
approve language changes

**Comment by jsubri** [comment]
I have tested this item :white_check_mark: successfully on 52a5aa21f2a74d4e249f04316e6360309b5f4a8e<br /><br />Test ok on upgraded site.


<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/22530">issues.joomla.org/tracker/joomla-cms/22530</a>.</sub>

**Comment by jsubri** [comment]
Note for testers: the below manual actions are required on upgraded site 3.9RC

insert into `#__action_logs_extensions` values (18, 'com_checkin');
delete from #__extensions where extension_id = 36
insert into `#__extensions` (`extension_id`, `package_id`, `name`, `type`, `element`, `folder`, `client_id`, `enabled`, `access`, `protected`, `manifest_cache`, `params`, `custom_data`, `system_data`, `checked_out`, `checked_out_time`, `ordering`, `state`) VALUES
(36, 0, 'com_actionlogs', 'component', 'com_actionlogs', '', 1, 1, 1, 1, '', '{"ip_logging":0,"csv_delimiter":",","loggable_extensions":["com_banners","com_cache","com_categories","com_checkin","com_config","com_contact","com_content","com_installer","com_media","com_menus","com_messages","com_modules","com_newsfeeds","com_plugins","com_redirect","com_tags","com_templates","com_users"]}', '', '', 0, '0000-00-00 00:00:00', 0, 0);

screen shot 2018-10-13 at 21 29 24<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/22530">issues.joomla.org/tracker/joomla-cms/22530</a>.</sub>

File: plugins/actionlog/joomla/joomla.php [hunk 3]
```diff
@@ -889,4 +889,40 @@ public function onUserAfterRemind($user)
...
+	/**
+	 * On after Check-in request
+	 *
+	 * Method is called after user request to check-in items.
+	 *
+	 * @param   array  $table  Holds the table name.
+	 *
+	 * @return  void
+	 *
+	 * @since   3.9.0
```
Comment (Quy) [inline plugins/actionlog/joomla/joomla.php:902]: Change to `__DEPLOY_VERSION__`

File: installation/sql/postgresql/joomla.sql [hunk 4]
```diff
@@ -2172,9 +2172,10 @@ INSERT INTO "#__action_logs_extensions" ("id", "extension") VALUES
 (14, 'com_redirect'),
 (15, 'com_tags'),
 (16, 'com_templates'),
-(17, 'com_users');
+(17, 'com_users'),
+(18, 'com_checkin';
```
Comment (Quy) [inline installation/sql/postgresql/joomla.sql]: Add `)`

File: plugins/actionlog/joomla/joomla.php (hunk 3 above)
Comment (alikon) [reply plugins/actionlog/joomla/joomla.php:902]: done

File: installation/sql/postgresql/joomla.sql (hunk 4 above)
Comment (alikon) [reply installation/sql/postgresql/joomla.sql]: done

**Comment by jsubri** [comment]
What about @mbabker suggestion to have
insert into #__action_logs_extensions values (18, 'com_checkin');
In a separated file ?

File: plugins/actionlog/joomla/joomla.php (hunk 3 above)
Comment (Quy) [reply plugins/actionlog/joomla/joomla.php:902]: The change should have been done to line 902.

**Comment by twister65** [comment]
After this patch is applied, the tables in the database are not updated.
com_checkin is missing in `#__extensions` and `#__action_logs_extensions` tables.

**Comment by alikon** [comment]
true, this needs some more cares, when it was proposed 3.9 wasn't released yet...
on my todo  list

**Comment by alikon** [comment]
redone with #23526

**Pull request closed without merging**